config/externalname.go

```diff
@@ -15,9 +15,38 @@ import (
 	"github.com/upbound/provider-aws/config/common"
 )
 
+// TerraformPluginFrameworkExternalNameConfigs contains all external
+// name configurations belonging to Terraform Plugin Framework
+// resources to be reconciled under the no-fork architecture for this
+// provider.
+var TerraformPluginFrameworkExternalNameConfigs = map[string]config.ExternalName{
+	// ec2
+	//
+	// Imported by using the id: sgr-02108b27edd666983
+	"aws_vpc_security_group_egress_rule": vpcSecurityGroupRule(),
+	// Imported by using the id: sgr-02108b27edd666983
+	"aws_vpc_security_group_ingress_rule": vpcSecurityGroupRule(),
+
+	// cognito
+	//
+	// us-west-2_abc123/3ho4ek12345678909nh3fmhpko
+	"aws_cognito_user_pool_client": cognitoUserPoolClient(),
+
+	// simpledb
+	//
+	// SimpleDB Domains can be imported using the name
+	"aws_simpledb_domain": config.NameAsIdentifier,
+
+	// appconfig
+	//
+	// AppConfig Environments can be imported by using the environment ID and application ID separated by a colon (:)
+	// terraform-plugin-framework
+	"aws_appconfig_environment": appConfigEnvironment(),
```

Comment:
If I'm not mistaken, the external-name syntax is not changing for the resource as we move from the CLI-based implementation to the native fw reconciliation. We are allowed to break APIs with the next release, and if we are breaking APIs, it's important for us to capture this in the release notes.